src/aead/less_safe_key.rs

```diff
@@ -74,12 +94,27 @@ impl LessSafeKey {
     where
         A: AsRef<[u8]>,
     {
+        let ciphertext_and_tag_len = in_out
+            .len()
+            .checked_sub(ciphertext_and_tag.start)
+            .ok_or(error::Unspecified)?;
+
+        let ciphertext_len = ciphertext_and_tag_len
+            .checked_sub(TAG_LEN)
+            .ok_or(error::Unspecified)?;
+
+        let (in_out, received_tag) = in_out.split_at_mut(ciphertext_and_tag.start + ciphertext_len);
+
+        let received_tag = (*received_tag).try_into()?;
+
         open_within_(
             self,
             nonce,
             Aad::from(aad.as_ref()),
+            received_tag,
             in_out,
             ciphertext_and_tag,
```

Comment:
Instead of passing `ciphertext_and_tag` here, pass in `in_out` shortened by the length of the tag.

Reply:
Thanks for the suggestion, I've made the change.